src/main/scala/net/snowflake/spark/snowflake/Conversions.scala

```diff
@@ -193,7 +196,23 @@ private[snowflake] object Conversions {
     * Parse a string exported from a Snowflake TIMESTAMP column
     */
   private def parseTimestamp(s: String, isInternalRow: Boolean): Any = {
+    // Need to handle the nano seconds filed separately
```

Comment:
Could you please add an internal parameter to enable/disable the change? It's enabled by default.
If some users is broken, they can disable the fix as a workaround.
For example, https://github.com/snowflakedb/spark-snowflake/blob/master/src/main/scala/net/snowflake/spark/snowflake/Parameters.scala#LL163C14-L163C14

Reply:
hmm.. do you mean to have a parameter such as `internal_support_micro_second_during_unload` ?  IMO, we should always try to support micro second level precision since by default direct JDBC supports this precision. It would make this part of the code confusing if we have two code paths and two set of timestamp patterns..

Reply:
I agree users need to use this fix. This is why we can set the internal parameter as true by default.
This is the internal policy to introduce a parameter to disable the fix if possible.
The internal parameter can be removed later. So the code will be clean up accordingly.

Reply:
Thanks for the suggestion. Will push a new commit to add this parameter.